### DISCUSSION TITLE
Enable admin page

### DISCUSSION THREAD

**Discussion by FlyingPersian** [dani-garcia/vaultwarden#4066] [Q&A] [answered]
Hello,

while changing a setting in the admin page, I unintentionally disabled the admin page by removing the admin token that was shown with stars on the admin page. I thought bitwarden just auto-filled the password field, so I didn't want it to be changed. I now cannot log in with the password. I read in the [wiki ](https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#disabling-the-admin-page)that removing the password actuall disables the admin page. I can confirm that the config.json does not have a ADMIN_TOKEN entry. 

How can I re-enable it now? I'm running vaultwarden in a docker swarm environment using portainer stacks, so basically docker_compose. I restarted the stack multiple times, didn't help. I didn't want to just add the admin_token to the config.json. I suppose that adding "admin_token": "xyz" would re-enable it? I'm not even sure if the position in the config matters. Is there a cleaner way to do it?

**Comment by netboy3** [comment]
You can try setting the ADMIN_TOKEN environment variable when starting the container. As long as you don't have an admin_token{} block, it should stick. You can then set it up in the GUI which will override the environment, then you can remove the environment. Please note that the recommendation is to either use environment or use the GUI (config.json), but not both. Read the wiki page you quoted above to see how to setup the variable and how to compute the hash.

**Reply by FlyingPersian** [reply]
Thanks! I actually restored an older version of the config, I forgot that I had duplicati running.

In docker-compose I'm actually using a argon2 hashed password. When I want to use that in the config.json I should use the GUI to do so. How would I input that? I tried with the full $$argon2.... But when I did that and tried to log in using the password in plain text, it didn't work.

**Reply by netboy3** [reply, marked as answer]
I never used the GUI and have always set everything in the docker-compose environment, but you should probably paste the argon2 string without the `$` escapes (I.E. reduce all the `$$` to `$`). If in doubt, just generate a new argon2 string as per the instructions in: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#using-argon2

**Reply by FlyingPersian** [reply]
Ohh I had to reduce all the $$ to $. That fixed it, thank you!
